unanet-summarizer.js

```diff
@@ -1,142 +1,156 @@
+
+        var bottomHeaderRow = document.createElement('tr');
+
+        var hoursHeading = '';
+        props.hoursByProjectType.forEach(function(timeItem){
+            hoursHeading = hoursHeading + '<th>' + timeItem.projectType + '</th>';
+        });
+
+        bottomHeaderRow.innerHTML = bottomHeaderRow.innerHTML + hoursHeading;
+        bottomHeaderRow.innerHTML = bottomHeaderRow.innerHTML + '<th>+ Hours</th><th>Non + Hours</th><th>Grand Total</th>';
+
+        var tableBody = document.createElement('tbody');
+        var dataRow = document.createElement('tr');
+
+        props.hoursByProjectType.forEach(function(timeItem){
+            var newCell = dataRow.insertCell(-1);
+            newCell.textContent = timeItem.totalHours;
+        });
+
+        var totalPlusCell = dataRow.insertCell(-1);
+        var totalNonPlusCell = dataRow.insertCell(-1);
+        var totalHoursCell = dataRow.insertCell(-1);
+
+        totalPlusCell.textContent = props.totalPlusHoursResult;
+        totalNonPlusCell.textContent = props.totalNonPlusHoursResult;
+        totalHoursCell.textContent = props.totalHoursResult;
+
+        tableBody.appendChild(dataRow);
+
+        tableHeader.appendChild(topHeaderRow);
+        tableHeader.appendChild(bottomHeaderRow);
+
+        resultTable.appendChild(tableHeader);
+        resultTable.appendChild(tableBody);
+
+        return resultTable;
+    };
+
+    var createContainer = function() {
+        var container = document.createElement('div');
+        container.id = CONTAINER_ID;
+        container.style = 'margin-bottom: 20px;';
+        return document.body.insertBefore(container, document.body.firstChild);
+    };
+    // Execution 
+    var getContainer = function() {
+        return document.getElementById(CONTAINER_ID) || createContainer();
+    };
+
+    var getReducers = function() {
+        return {
+            hoursByProjectType: [ totalHoursByProjectType, [] ],
+            totalPlusHoursResult: [ totalPlusHours, 0.0 ],
+            totalNonPlusHoursResult: [ totalNonPlusHours, 0.0 ],
+            totalHoursResult: [ totalHoursReduceFunction, 0.0 ]
+        };
+    };
     
-    return acc;
-};
-
-var totalPlusHours = function(acc, obj){
-    if (obj.projectType.includes("+")){
-        acc = acc + obj.timeValue;
-    }
-    return acc;
-};
-
-var totalNonPlusHours = function(acc, obj){
-    if (!obj.projectType.includes("+")){
-        acc = acc + obj.timeValue;
-    }
-    return acc;
-};
-
-var docTemplateGeneration = function(hoursByProjectTypeArray, totalPlus, totalNonPlus, totalHours){
-
-    var resultTable = document.createElement('table');
-    resultTable.setAttribute('style', 'border: 2px solid;');
+    return function() { 
+        var timeEntries = obtainTimeEntryRows();        
+        var reducers = getReducers();
 
-    var tableHeader = document.createElement('thead');
+        var properties = Object.keys(reducers).reduce(function(acc, property) {
```

Comment:
@jmweir do you think this impacts the readability? Maybe I'm just not in JS enough, but it reads a little cryptic to me.

Reply:
I think readability is a good callout. This can be made more readable, will submit with next PR.